### DISCUSSION TITLE
How to generate a changelog?

### DISCUSSION THREAD

**Discussion by holtetx** [Zhengqbbb/cz-git#169] [Q&A] [answered]
Hi!

How can I generate a changelog when i've used the `czg` cli? 
We're using some custom scopes, and tools like [standard-changelog](https://www.npmjs.com/package/standard-changelog#cli) and [conventional-changelog-cli](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) does not seem to recognize the commits with custom scopes. How can i 

1) Get detectable, standard, scopes into the czg cli? My custom scope can perhaps be moved to the commit body instead.
or
2) Create a changelog with custom scopes?

I run a global install of czg cli. Our projects run accross node, java, scala and react projects.

Our config is: 
```
const { execSync } = require('child_process');

// Get the current branch name
const branchName = execSync('git rev-parse --abbrev-ref HEAD').toString().trim();

// Extract the ETU-XXXXX part
const etuPartMatch = branchName.match(/ETU-\d+/);
const etuPart = etuPartMatch ? etuPartMatch[0] : null;

module.exports = {
  rules: {
    'subject-min-length': [2, 'always', 2],
    'subject-empty': [2, 'never']
  },
  prompt: {
    themeColorCode: '38;5;043',
    scopes: etuPart ? [{ name: etuPart }] : undefined,
    allowCustomScopes: true,
    allowEmptyScopes: true,
    skipQuestions: ['footer', 'footerPrefix']
 }
};
```

Thank you in advance!

**Comment by Zhengqbbb** [comment, marked as answer]
> tools like [standard-changelog](https://www.npmjs.com/package/standard-changelog#cli) and [conventional-changelog-cli](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) does not seem to recognize the commits with custom scopes.

It shouldn't be. `czg` follows the standard [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
So czg can be recognized by most changelog tools
https://github.com/Zhengqbbb/cz-git/blob/main/CHANGELOG.md

---
https://github.com/Zhengqbbb/cz-git/blob/56f91afd8eb37654cc3151385fce2c1c57b095b7/package.json#L44

https://github.com/Zhengqbbb/cz-git/blob/56f91afd8eb37654cc3151385fce2c1c57b095b7/.x-cmd/release#L15-L19

**Reply by holtetx** [reply]
Thanks for your reply! How can i populate the scopes with common, recognizable scopes? I'll put our custom tag as a part of the commit message instead.

EDIT: I believe what broke our changelogs were the use of hyphens in the scope? Somehow that did not work, but when i manually added scopes like 'config', 'frontend', 'server', ut seemed to work. Or perhaps it was due to XXXX in being numbers. A tag would look like `ETU-12345`.
